src/lib/icu/meson.build

```diff
@@ -18,8 +18,18 @@ if icu_dep.found()
     'Init.cxx',
   ]
 elif not get_option('iconv').disabled()
-  have_iconv = compiler.has_function('iconv', prefix : '#include <iconv.h>')
-  conf.set('HAVE_ICONV', have_iconv)
+  have_iconv = c_compiler.has_header_symbol('iconv.h', 'iconv')
+  if have_iconv
+    libiconv = c_compiler.find_library('iconv', required : false)
+    conf.set('HAVE_ICONV', have_iconv)
+    libiconv_dep = static_library('iconv', icu_sources, include_directories: inc, dependencies: libiconv)
```

Comment:
This compiles `icu_sources` twice.

Reply:
I *think* I fixed.

Reply:
Now the `icu_sources` are only compiled when an external libiconv is used. Not when libc's `iconv()` is used and not when libicu is used.